OTAnalytics/domain/event.py

```diff
@@ -1,2 +1,177 @@
+        greater equal one.
+        road_user_type (str): the road user type involved with this event.
+        hostname (str): the OTCamera hostname that the video is associated with.
+        occurrence (datetime): the time when this event occurred.
+        frame_number (int): the video frame number that this event is associated with.
+        section_id (Optional[str]): only set when event type is of section.
+        Defaults to `None`.
+        event_coordinate (ImageCoordinate): location where the event occurred in
+        the video.
+        event_type (EventType): this event's type.
+        direction_vector (DirectionVector2D): a 2-dimensional direction vector denoting
+        the direction of the road user associated with this event.
+        video_name (str): the video name associated with this event.
+
+    """
+
+    road_user_id: int
+    road_user_type: str
+    hostname: str
+    occurrence: datetime
+    frame_number: int
+    section_id: Optional[str]
+    event_coordinate: ImageCoordinate
+    event_type: EventType
+    direction_vector: DirectionVector2D
+    video_name: str
+
+    def _validate(self) -> None:
+        self._validate_road_user_id_greater_zero()
+        self._validate_frame_number_greater_equal_one()
+
+    def _validate_frame_number_greater_equal_one(self) -> None:
+        if self.frame_number < 1:
+            raise (
+                ValueError(
+                    (
+                        "frame number must be greater equal one, "
+                        f"but it {self.frame_number}"
+                    )
+                )
+            )
+
+    def _validate_road_user_id_greater_zero(self) -> None:
+        if self.road_user_id < 0:
+            raise ValueError(
+                f"vehicle_id must be greater than zero, but is {self.road_user_id}"
+            )
+
+
+class EventBuilder(ABC):
+    @abstractmethod
+    def create_event(self, detection: Detection) -> Event:
+        pass
+
+    def extract_hostname(self, file_path: Path) -> str:
+        """Parse the given filename and retrieve the start date of the video.
+
+        Args:
+            video_file (Path): path to video file
+
+        Raises:
+            InproperFormattedFilename: if the filename is not formatted as expected, an
+            exception will be raised
+
+        Returns:
+            datetime: start date of the video
+        """
+        match = re.search(
+            FILE_NAME_PATTERN,
+            file_path.name,
+        )
+        if match:
+            hostname: str = match.group(HOSTNAME)
+            return hostname
+        raise InproperFormattedFilename(f"Could not parse {file_path.name}.")
+
+
+class SectionEventBuilder(EventBuilder):
+    def __init__(self) -> None:
+        self.section_id: Optional[str] = None
+        self.direction_vector: Optional[DirectionVector2D] = None
+        self.event_type: Optional[EventType] = None
+
+    def reset(self) -> None:
```

Comment:
Why do we need a reset method?
Can we not just create a new Builder?

Reply:
I did it just in case some one tries to reuse the builder. But of course we can just use a new one